state/protocol/badger/mutator_test.go

```diff
@@ -1081,38 +1126,32 @@ func TestExtendReceiptsValid(t *testing.T) {
 	util.RunWithProtocolStateAndMutatorFactory(t, mockMutatorFactory(), func(db *badger.DB, state *protocol.State) {
 		// bootstrap the root block
 		block1, result, seal := unittest.BootstrapFixture(participants)
-		block1.Payload.Guarantees = nil
-		block1.Header.PayloadHash = block1.Payload.Hash()
+		block1.SetPayload(flow.EmptyPayload())
 		err := state.Mutate().Bootstrap(block1, result, seal)
 		require.NoError(t, err)
 
-		// create block2 and block3
 		block2 := unittest.BlockWithParentFixture(block1.Header)
-		block2.Payload.Guarantees = nil
-		block2.Header.PayloadHash = block2.Payload.Hash()
+		block2.SetPayload(flow.EmptyPayload())
 		err = state.Mutate().Extend(&block2)
 		require.Nil(t, err)
 
 		block3 := unittest.BlockWithParentFixture(block2.Header)
-		block3.Payload.Guarantees = nil
-		block3.Payload.Receipts = append(block3.Payload.Receipts, unittest.ReceiptForBlockFixture(&block2))
-		block3.Header.PayloadHash = block3.Payload.Hash()
+		block3.SetPayload(flow.EmptyPayload())
 		err = state.Mutate().Extend(&block3)
 		require.Nil(t, err)
 
 		block4 := unittest.BlockWithParentFixture(block3.Header)
-		block4.Payload.Guarantees = nil
-		block4.Header.PayloadHash = block4.Payload.Hash()
+		block4.SetPayload(flow.EmptyPayload())
 		err = state.Mutate().Extend(&block4)
 		require.Nil(t, err)
 
 		block5 := unittest.BlockWithParentFixture(block4.Header)
-		block5.Payload.Guarantees = nil
-		block5.Payload.Receipts = append(block5.Payload.Receipts,
-			unittest.ReceiptForBlockFixture(&block3),
-			unittest.ReceiptForBlockFixture(&block4),
-		)
-		block5.Header.PayloadHash = block5.Payload.Hash()
+		block5.SetPayload(flow.Payload{
+			Receipts: []*flow.ExecutionReceipt{
+				unittest.ReceiptForBlockFixture(&block3),
+				unittest.ReceiptForBlockFixture(&block4),
+			},
+		})
```

Comment:
would be great to also cover:
* second receipts for a block (e.g `block3`), which commits to a _different_ result
* another receipts for a block (e.g `block3`), which commits to a _the same_ result (i.e. just the executor differs)

In total, this could look like
   ```golang
		receipt3a := unittest.ReceiptForBlockFixture(&block3)
		receipt3b := unittest.ReceiptForBlockFixture(&block3)
		var altResult flow.ExecutionResult = receipt3b.ExecutionResult // copy
		altResult.Signatures = unittest.SignaturesFixture(1)
		receipt3c := unittest.ExecutionReceiptFixture(unittest.WithResult(&altResult))

		block5 := unittest.BlockWithParentFixture(block4.Header)
		block5.SetPayload(flow.Payload{
			Receipts: []*flow.ExecutionReceipt{
				receipt3a,
				receipt3b,
				receipt3c,
				unittest.ReceiptForBlockFixture(&block4),
			},
		})
   ```